lisa/core/node.py

```diff
@@ -98,6 +100,19 @@ def setConnectionInfo(
         self.internalAddress = address
         self.internalPort = port
 
+    def getScriptPath(self) -> pathlib.Path:
+        assert self.workingPath
```

Comment:
We're mixing use of asserts and exceptions. That can be reasonable, but we should document why and when we choose one over the other.

Reply:
The assert uses for Nonable objects.  Once it's in use, as code flow, it's impossible to be None, but static type checking prompt possibility to be None. Add this line to avoid checking error. The static type checking need to declare variables in __init__, but I don't want to init a real object there. So I have to give it Optional[].

There are some above instances are using Exception, I will have a clean up.

assert is shorter, but it has no message, so it shouldn't be used with complex logic. Assert Nonable is straightforward.